core/src/test/java/org/hibernate/ogm/utils/TestHelper.java

```diff
@@ -289,17 +295,37 @@ private static AvailableDatastoreProvider getDatastoreProvider() {
 			// This ignores the case where the provider is given as class or FQN; That's ok for now, can be extended if
 			// needed
 			Object datastoreProviderProperty = getDefaultTestStandardServiceRegistry( Collections.<String, Object>emptyMap() )
-				.getService( ConfigurationService.class )
-				.getSettings()
-				.get( OgmProperties.DATASTORE_PROVIDER );
-
-			AvailableDatastoreProvider provider = datastoreProviderProperty != null ? AvailableDatastoreProvider.byShortName( datastoreProviderProperty.toString() ) : null;
-
-			if ( provider == null ) {
+					.getService( ConfigurationService.class )
+					.getSettings()
+					.get( OgmProperties.DATASTORE_PROVIDER );
+
+			if ( datastoreProviderProperty != null ) {
+				String value = datastoreProviderProperty.toString();
+				if ( AvailableDatastoreProvider.isShortName( value ) ) {
+					return AvailableDatastoreProvider.byShortName( value );
+				}
+
+				for ( AvailableDatastoreProvider provider : AvailableDatastoreProvider.values() ) {
+					if ( provider.getDatastoreProviderClassName().equals( value ) ) {
+						return provider;
+					}
+				}
+
+				Class<?> configuredProviderClass = loadClass( value );
+				if ( configuredProviderClass != null ) {
+
+					for ( AvailableDatastoreProvider provider : AvailableDatastoreProvider.values() ) {
+						Class<?> availableProviderClass = loadClass( provider.getDatastoreProviderClassName() );
+
+						if ( availableProviderClass.isAssignableFrom( configuredProviderClass ) ) {
```

Comment:
I extended the functionality so it's no longer limited to short names but allows exact type matching and to exclude whole hierarchies (subtypes of providers). Will split up the method into parts.